src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -4,22 +4,467 @@ Thomas Segismont
+  ],
+  "availability": [
+    {
+      "id": "server1",
+      "data": [
+        {"timestamp": 1460111065369, value: "up"},
+        {"timestamp": 1460151065369, value: "up"}
+      ]
+    },
+    {
+      "id": "server2",
+      "data": [
+        {"timestamp": 1460111065369, value: "unknown"},
+        {"timestamp": 1460151065369, value: "up"}
+      ]
+    }
+  ]
+}
+----
+
+== Data Retention and Removal
+Metric data is automatically deleted from the system after an amount of time that is determined by data retention
+settings. Data retention can be specified at various levels and is specified in days. There is a system-wide default of
+seven days. This setting will apply to all metrics in the system if no other settings are specified. The system-wide
+setting can be overridden at start up by either setting the `hawkular.metrics.default-ttl` system property or by
+setting the `DEFAULT_TTL` environment variable.
+
+Data retention can also be set per tenant. To do this, you need to explicitly create the tenant as in the following
+example.
+
+[source,shell]
+----
+curl -X POST http://server/hawkular/metrics/tenants -d @payload.json \
+-H "Content-Type: application/json"
+----
+
+[source,javascript]
+----
+{
+  "id": "com.acme",
+  "retentions": {
+    "gauge": 10,
+    "counter": 5,
+    "availability": 8
+  }
+}
+----
+
+This example uses the curl shell command. The request body is put in a file to improve readability. The `retentions`
+map consists of names of one or more metric types. The value of each is an integer which represents the data retention
+for that metric type in days.
+
+You can also set data retention at the individual metric level. This would override any tenant data retention as well
+as the system-wide default. Here is an example.
+
+[source,shell]
+----
+curl -X POST http://server/hawkular/metrics/metrics -d @payload.json \
+-H "Content-Type: application/json" -H "Hawkular-Tenant: com.acme"
+----
+
+[source,javascript]
+----
+{
+  "id": "request_size",
+  "dataRetention": 10
+}
+----
+
+This request creates a gauge named `request_size` with a data retention of 10 days.
+
+WARNING: Hawkular Metrics currently lacks APIs for changing data retention. See
+https://issues.jboss.org/browse/HWKMETRICS-380[HWKMETRICS-380] for details.
+
+*TODO*
+Add section on how Cassandra handles deletes.
+
+== Tagging
+Tags in Hawkular Metrics are key/value pairs. Tags can be applied to a metric to provide meta data for the time series
+as a whole. Tags can also be applied to individual data points. Tags can be used to perform filtering in queries.
+
+=== Creating Metric Tags
+[source,shell]
+----
+curl -X POST http://server/hawkular/metrics/gauges -d @payload.json \
+-H "Content-Type: application/json" -H "Hawkular-Tenant: com.acme"
+----
+
+[source,javascript]
+----
+{
+  "id": "request_size",
+  "tags": {
+    "datacenter": "dc1",
+    "env": "stage"
+    "units": "bytes"
+  }
+}
+----
+
+There are similar endpoints for other metric types.
```

Comment:
It might make sense to tell the user how the behavior of update & delete differ a bit. Any update will replace all the stored tags. So if the idea was to replace a single key/value pair's value, then all the tags must be defined in the request, otherwise the rest are deleted.

With delete however, only the given value is deleted (as assumed). The difference here is something that certainly makes sense, but might not be obvious to everyone (especially in case one wants to update a single value).